### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by harbournick**
Adds (optional) pagination v1 to SuperEditor

### CODE REVIEW

File: packages/super-editor/src/core/config/style.js [hunk 1]
```diff
@@ -101,6 +101,16 @@ img.ProseMirror-separator {
   margin-bottom: 1.5px;
 }
 
+.ProseMirror {
+  width: 100%;
+  outline: none;
+  bordeR: none;
```
Comment (artem-harbour) [inline packages/super-editor/src/core/config/style.js:107]: `border`

File: packages/super-editor/src/extensions/pagination/pagination.js [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    let shouldUpdate = false;
+
+    // Track wether the first load has occured or not
+    let hasInitialized = false;
+
+    const paginationPlugin = new Plugin({
+      key: paginationPluginKey,
+      state: {
+        init() {
+          return DecorationSet.empty;
```
Comment (artem-harbour) [inline packages/super-editor/src/extensions/pagination/pagination.js:37]: Probably here it is necessary to have prepared decorations right away.

File: packages/super-editor/src/extensions/pagination/pagination.js [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+    footerData,
+    location,
+    sectionHeight,
+    suppressFooter,
+    suppressHeader,
+    editor,
+  };
+
+  const header = createApp({
+    render: () => h(PageBreak, { data, coords, sectionHeight, styles, isDebugging })
```
Comment (artem-harbour) [inline packages/super-editor/src/extensions/pagination/pagination.js:325]: I would not recommend using this method for many reasons.

- For each such case, Vue app will be created, which can be very expensive (including decorations, all recalculations and so on).
- Vue render functions are less optimized than regular components (and are more suitable for small UI).
- Dependency on Vue in PM editor.

In the case of PM it is always preferable to use NodeViews, decorations, native work with DOM.

**Comment by harbournick** [comment]
Closing this one as making major changes to the approach to remove createApp etc

**Pull request closed without merging**